api/configuration_default.go

```diff
@@ -0,0 +1,44 @@
+package api
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net/http"
+	"net/url"
+)
+
+type ConfigParams struct {
+	Host             *url.URL
+	UserAgent        string
+	Token            *string
+	TraceId          *string
+	AllowInsecureTLS bool
+	Debug            bool
+}
+
+// NewAPIConfig builds a configuration tailored to the InfluxDB v2 API.
+func NewAPIConfig(params ConfigParams) *Configuration {
+	clientTransport := http.DefaultTransport.(*http.Transport)
+	clientTransport.TLSClientConfig = &tls.Config{InsecureSkipVerify: params.AllowInsecureTLS}
```

Comment:
Since this is likely the only use in the process it's probably fine, but should we be mutating the global DefaultTransport?

Reply:
Updated to `Clone()` the transport first.